### PR TITLE
feat: add otel_scope_info and scope labels

### PR DESCRIPTION
**Pull request by zengxilong**
PR for #930 

1. Add otel_scope_name and otel_scope_version attributes to all metrics based on the instrumentation scope name and version.
2. Add an otel_scope_info metric.
3. Add prometheus exporter config, provide a disable_scope_info flag to disable the otel_scope_info metric and otel_scope_ labels.
4. I will make a new PR to resolve collisions between TYPE, HELP, and UNIT comments.

### CODE REVIEW

File: opentelemetry-prometheus/src/lib.rs [hunk 1]
```diff
@@ -175,6 +204,22 @@ impl ExporterBuilder {
...
+#[derive(Debug, Default)]
+pub struct ExporterConfig {
+    /// disable the otel_scope_info metric and otel_scope_ labels.
+    disable_scope_info: bool,
+}
+
+impl ExporterConfig {
+    /// Set disable_scope_info for [`ExporterConfig`].
+    /// It's the flag to disable the otel_scope_info metric and otel_scope_ labels.
+    pub fn disable_scope_info(mut self) -> Self {
```
Comment (TommyCpp) [inline opentelemetry-prometheus/src/lib.rs:217]: I know we should default to `false` here but still may be worth to allow the user to pass a bool to change the behavior

**Review by TommyCpp — COMMENTED** [review]
Thanks for working on this. Overall like a good start. I think maybe use something like `with_scope_info` will be more clear. And we can have the default value to be `true`

File: opentelemetry-prometheus/src/lib.rs (hunk 1 above)
Comment (zengxilong) [reply opentelemetry-prometheus/src/lib.rs:217]: So does `otel_scope_info` need to be disabled by default?

Comment (hdost) [reply opentelemetry-prometheus/src/lib.rs:217]: So based on the spec it should be on by default https://github.com/open-telemetry/opentelemetry-specification/blob/0d9081669c0c087c092bf35fe5c5652c2b505b94/specification/compatibility/prometheus_and_openmetrics.md#instrumentation-scope-1
but I agree that  `with_scope_info`  would be  more inline with how the interfaces are working.

Comment (zengxilong) [reply opentelemetry-prometheus/src/lib.rs]: > So based on the spec it should be on by default https://github.com/open-telemetry/opentelemetry-specification/blob/0d9081669c0c087c092bf35fe5c5652c2b505b94/specification/compatibility/prometheus_and_openmetrics.md#instrumentation-scope-1 but I agree that `with_scope_info` would be more inline with how the interfaces are working.

I changed `disable_scope_info(mut self) -> Self` to `with_scope_info(mut self, enabled: bool) -> Self`, and let the default value be true.

**Review by hdost — APPROVED** [review]

**Pull request merged**